test/val/val_instructions_test.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2017 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Validation tests for illegal instructions
+
+#include <sstream>
+#include <string>
+#include <tuple>
+
+#include "gmock/gmock.h"
+#include "val_fixtures.h"
+
+using ::testing::HasSubstr;
+
+using ValidateIns = spvtest::ValidateBase<std::string>;
+
+namespace {
+
```

Comment:
Makes sense, I refactored it to use TEST_P and INSTANTIATE_TEST_CASE_P now that I understand what they do.

Right now I'm using a std::pair as the test parameter and the operands are represented just as a second string.  Not sure if I should have made it a vector that I join, but a second string seems to work well enough.  There's also a bit of duplication left as the first 3 operands are the same for each case, but I feel like it would be less readable to move those out.